pkg/syncer/shared/namespace.go

```diff
@@ -105,3 +106,22 @@ func PhysicalClusterNamespaceName(l NamespaceLocator) (string, error) {
 	// keep the namespaces short enough.
 	return fmt.Sprintf("kcp-%s", base36hash[:12]), nil
 }
+
+// IndexByNamespaceLocator is a cache.IndexFunc that indexes namespaces by the namespaceLocator annotation.
+func IndexByNamespaceLocator(obj interface{}) ([]string, error) {
```

Comment:
If this is exposed for other users, I'd expect a constant for the name of the index as well as a function to format the key as well